src/node/snapshot_serialise.h

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "ds/logger.h"
+#include "ds/serialized.h"
+#include "entities.h"
+#include "kv/kv_types.h"
+#include "kv/serialised_entry_format.h"
+#include "node/nodes.h"
+#include "node/service.h"
+#include "node/tx_receipt.h"
+
+#include <nlohmann/json.hpp>
+
+namespace ccf
+{
+  struct StartupSnapshotInfo
+  {
+    std::vector<uint8_t>& raw;
+    kv::Version seqno;
+    std::optional<kv::Version> evidence_seqno = std::nullopt;
+
+    // Store used to verify a snapshot (either created fresh when a node joins
+    // from a snapshot or points to the main store when recovering from a
+    // snapshot)
+    std::shared_ptr<kv::Store> store = nullptr;
+
+    // The snapshot to startup from (on join or recovery) is only valid once a
+    // signature ledger entry confirms that the snapshot evidence was
+    // committed
+    bool has_evidence = false;
+    bool is_evidence_committed = false;
+
+    StartupSnapshotInfo(
+      const std::shared_ptr<kv::Store>& store_,
+      std::vector<uint8_t>& raw_,
+      kv::Version seqno_,
+      std::optional<kv::Version> evidence_seqno_) :
+      raw(raw_),
+      seqno(seqno_),
+      evidence_seqno(evidence_seqno_),
+      store(store_)
+    {}
+
+    bool is_snapshot_verified() const
+    {
+      return has_evidence && is_evidence_committed;
+    }
+
+    bool requires_ledger_verification() const
+    {
+      // Snapshot evidence seqno is only set by the host for 1.x snapshots
+      // whose evidence need to be verified in the ledger suffix on startup
+      return evidence_seqno.has_value();
+    }
+
+    ~StartupSnapshotInfo()
+    {
+      raw.clear();
+      raw.shrink_to_fit();
+    }
+  };
+
+  static void deserialise_snapshot(
```

Comment:
Since this contains a deserialisation, perhaps a rename to snapshot_serdes or something would be helpful.